### PR TITLE
RHOAIENG-1044: Update hardcoded constants as environment variables 

### PR DESCRIPTION
**Pull request by mholder6**
## Description

Refactored the controllers/utils function GetIstioControlPlaneName to no longer reference IstioNamespace and IstioControlPlaneName constants, and to return errors instead of automatically setting the env vars. Removed the constants from the constants/constants.go file, and replaced all references of the constants in the code with a call to the utils function mentioned before. 

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by VedantMahabaleshwarkar** [comment]
@mholder6 the changelist on this PR looks wrong, needs a rebase probably

File: Containerfile [hunk 1]
```diff
@@ -27,10 +23,6 @@ RUN CGO_ENABLED=0 GOOS=linux go build -a -o manager main.go
 FROM registry.access.redhat.com/ubi8/ubi-minimal:8.6
 WORKDIR /
 COPY --from=builder /workspace/manager .
-COPY --from=builder /workspace/metrics_dashboards/ovms-metrics.json .
```
Comment (spolti) [inline Containerfile:30]: Guess you need to rebase your PR.

Comment (spolti) [reply Containerfile:30]: a lot of unrelated changes.

File: controllers/utils/utils.go [hunk 2]
```diff
@@ -186,8 +186,8 @@ func GetIstioControlPlaneName(ctx context.Context, cli client.Client) (istioCont
 		log.V(1).Info("Trying to read Istio Control Plane name and namespace from DSCI")
 		objectList, err := getDSCIObject(ctx, cli)
 		if err != nil {
-			log.V(0).Error(err, "Failed to fetch the DSCI object, using default values")
-			return constants.IstioControlPlaneName, constants.IstioNamespace
```
Comment (israel-hdez) [inline controllers/utils/utils.go:190]: For the time being, it may be better to invoke `panic`, rather than returning the error string in the namespace return value.

Comment (mholder6) [reply controllers/utils/utils.go:190]: got it, thanks for the input!!

File: controllers/utils/utils.go [hunk 3]
```diff
@@ -19,8 +19,6 @@ import (
 	"k8s.io/client-go/discovery"
 	"k8s.io/client-go/rest"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-
-	"github.com/opendatahub-io/odh-model-controller/controllers/constants"
```
Comment (hdefazio) [inline controllers/utils/utils.go:23]: I think you need to add this back

Comment (mholder6) [reply controllers/utils/utils.go:23]: The import was no longer being used in the file -- still should be added back?

Comment (hdefazio) [reply controllers/utils/utils.go:29]: I think after rebasing on the latest master it will need the import again

File: controllers/constants/constants.go [hunk 4]
```diff
@@ -18,8 +18,6 @@ package constants
 const (
 	InferenceServiceKind = "InferenceService"
 
-	IstioNamespace                   = "istio-system"
```
Comment (spolti) [inline controllers/constants/constants.go:21]: Note that it is still used in case the func `utils.GetIstioControlPlaneName`

can't find anything: https://github.com/mholder6/odh-model-controller/blob/7a14b2092913e57cc2367ce14c0fa7787b2b6d1e/controllers/utils/utils.go#L217C6-L217C46

same for `IstioControlPlaneName`

It can't be deleted, you need to override where these values are hardcoded to the values returned from this function.

File: controllers/utils/utils.go [hunk 5]
```diff
@@ -213,8 +212,8 @@ func GetIstioControlPlaneName(ctx context.Context, cli client.Client) (istioCont
 					meshNamespace = namespace
 				} else {
 					log.V(1).Info("Mesh Namespace is not set in DSCI")
-					// at this point, it is not set anywhere, lets just use the default
-					meshNamespace = constants.IstioNamespace
+					// at this point, it is not set anywhere, lets return an error
```
Comment (spolti) [inline controllers/utils/utils.go:215]: it is better to return the default instead panicking, no @israel-hdez ?

File: controllers/constants/constants.go (hunk 4 above)
Comment (mholder6) [reply controllers/constants/constants.go:21]: Yes I actually am invoking panic instead of defaulting to those values -- you can see this here:
https://github.com/opendatahub-io/odh-model-controller/pull/254/files#diff-d93cfa921e60384e543609474ebc28fdbad7ae20f135f76398ef8d821333b3f2R194

File: controllers/utils/utils.go (hunk 3 above)
Comment (mholder6) [reply controllers/utils/utils.go:29]: thanks for this! I've rebased and pulled all the recent changes now. :)

File: controllers/utils/utils.go (hunk 2 above)
Comment (spolti) [reply controllers/utils/utils.go:196]: actually, it is logging an error and returning the default values, keeping backward capabilities.

File: controllers/utils/utils.go (hunk 5 above)
Comment (israel-hdez) [reply controllers/utils/utils.go:215]: My rationale here is that we shouldn't be using any defaults in the code. We should be forcing this to be specified.
Hoping I'm not wrong, it is always going to be set on the Deployment. If it is not set, the manifests would be faulty.

File: config/webhook/manifests.yaml [hunk 6]
```diff
@@ -9,7 +9,7 @@ webhooks:
   clientConfig:
     service:
       name: webhook-service
-      namespace: $(mesh-namespace)
+      namespace: system
```
Comment (israel-hdez) [inline config/webhook/manifests.yaml:12]: Can you revert this? It doesn't look related.

File: controllers/suite_test.go [hunk 7]
```diff
@@ -52,7 +52,7 @@ import (
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 
-	"github.com/opendatahub-io/odh-model-controller/controllers/constants"
+	//"github.com/opendatahub-io/odh-model-controller/controllers/constants"
```
Comment (israel-hdez) [inline controllers/suite_test.go:55]: Remove comment.

File: config/webhook/manifests.yaml (hunk 6 above)
Comment (spolti) [reply config/webhook/manifests.yaml:12]: Yeah. it is updated every time the manifests are generated.
@Jooho fyi.

File: controllers/kserve_inferenceservice_controller_test.go [hunk 8]
```diff
@@ -192,16 +194,18 @@ var _ = Describe("The Openshift Kserve model controller", func() {
 			err = cli.Status().Update(ctx, deployedInferenceService)
 			Expect(err).NotTo(HaveOccurred())
 
+			_, meshName := utils.GetIstioControlPlaneName(ctx, cli)
```
Comment (spolti) [inline controllers/kserve_inferenceservice_controller_test.go:197]: please rename to meshNamespace to avoid confunsions.

File: controllers/kserve_inferenceservice_controller_test.go [hunk 9]
```diff
@@ -246,6 +250,7 @@ var _ = Describe("The Openshift Kserve model controller", func() {
 			ctx := context.Background()
 			testNamespace := Namespaces.Create(cli)
 			testNs = testNamespace.Name
+			_, meshName := utils.GetIstioControlPlaneName(ctx, cli)
```
Comment (spolti) [inline controllers/kserve_inferenceservice_controller_test.go:253]: same

File: controllers/kserve_inferenceservice_controller_test.go [hunk 10]
```diff
@@ -366,10 +372,12 @@ var _ = Describe("The Openshift Kserve model controller", func() {
 				Expect(err).NotTo(HaveOccurred())
 				Expect(cli.Delete(ctx, deployedInferenceService)).Should(Succeed())
 
+				_, meshName := utils.GetIstioControlPlaneName(ctx, cli)
```
Comment (spolti) [inline controllers/kserve_inferenceservice_controller_test.go:375]: here.

File: controllers/reconcilers/kserve_route_reconciler.go [hunk 11]
```diff
@@ -129,10 +131,12 @@ func (r *KserveRouteReconciler) createDesiredResource(isvc *kservev1beta1.Infere
 			}
 		}
 
+		_, meshName := utils2.GetIstioControlPlaneName(context.Background(), r.client)
+
 		route := &v1.Route{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:        getKServeRouteName(isvc),
-				Namespace:   constants2.IstioNamespace,
+				Namespace:   meshName,
```
Comment (spolti) [inline controllers/reconcilers/kserve_route_reconciler.go:139]: here as well.

File: controllers/suite_test.go [hunk 12]
```diff
@@ -140,10 +140,11 @@ var _ = BeforeSuite(func() {
 	Expect(cli).NotTo(BeNil())
 
 	// Create istio-system namespace
+	_, meshName := utils.GetIstioControlPlaneName(ctx, cli)
 	istioNamespace := &corev1.Namespace{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      constants.IstioNamespace,
-			Namespace: constants.IstioNamespace,
+			Name:      meshName,
```
Comment (spolti) [inline controllers/suite_test.go:146]: here as well.

File: controllers/utils/utils.go (hunk 5 above)
Comment (spolti) [reply controllers/utils/utils.go:215]: That make sense, but is this the kind of error we want to block the operator startup?

File: Makefile [hunk 13]
```diff
@@ -69,7 +69,7 @@ vet: ## Run go vet against code.
 
 .PHONY: test
 test: manifests generate fmt vet envtest ## Run tests.
-	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" POD_NAMESPACE=default go test -v ./controllers/... -coverprofile cover.out
+	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" POD_NAMESPACE=default MESH_NAMESPACE=istio-system CONTROL_PLANE_NAME=istio-system go test -v ./controllers/... -coverprofile cover.out
```
Comment (spolti) [inline Makefile:72]: Can you please elaborate on why is this needed?

Comment (mholder6) [reply Makefile:72]: I believe the test was not building correctly before making these changes, @israel-hdez do you remember specifically why this was necessary?

Comment (israel-hdez) [reply Makefile:72]: Since there is no default value anymore, running tests would fail if we don't set these envvars, because the code would see empty values.

File: controllers/utils/utils.go (hunk 5 above)
Comment (israel-hdez) [reply controllers/utils/utils.go:215]: As it is, this should not block operator startup. Although, it will crash during reconciliation of some ISVC.
I expect no crash happening as long as all configuration is correct.

But yes, we can block operator startup to fail early. The panics would work as a back-up. Probably better to do that in another PR to not complicate this one.

**Review by israel-hdez — APPROVED** [review]
It is looking good to me now.

I'll let @spolti to give the final approval, since he also provided some feedback.

File: controllers/utils/utils.go [hunk 14]
```diff
@@ -245,7 +244,8 @@ func VerifyIfMeshAuthorizationIsEnabled(ctx context.Context, cli client.Client)
 	// is available, taking into account both managed an unmanaged configurations.
 
 	authPolicies := &v1beta1.AuthorizationPolicyList{}
-	if err := cli.List(ctx, authPolicies, client.InNamespace(constants.IstioNamespace)); err != nil && !meta.IsNoMatchError(err) {
+	_, meshName := GetIstioControlPlaneName(ctx, cli)
+	if err := cli.List(ctx, authPolicies, client.InNamespace(meshName)); err != nil && !meta.IsNoMatchError(err) {
```
Comment (spolti) [inline controllers/utils/utils.go:248]: you missed some places like this :)

Comment (spolti) [reply controllers/utils/utils.go:248]: sry about asking you to rename, but it it makes things easier to read :)

Comment (mholder6) [reply controllers/utils/utils.go]: No problem, everything should be updated now :)

File: config/webhook/manifests.yaml (hunk 6 above)
Comment (spolti) [reply config/webhook/manifests.yaml:12]: @mholder6 can you please revert?

File: controllers/utils/utils.go (hunk 5 above)
Comment (spolti) [reply controllers/utils/utils.go:215]: Right, @mholder6 can you please open a follow-up Jira?

**Review by spolti — COMMENTED** [review]
Just one more thing and I think it is good to go.

**Review by spolti — APPROVED** [review]
